### PR TITLE
refactor: Update `master` branch mentioning to `main`

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

### What does this PR do?
Updates the unit tests, CI jobs and other mentions of `master` branch to `main`.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20000

### How to test this PR?

Build jobs should pass. At least one job for HappyPath tests should pass.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

### CODE REVIEW

File: generator/src/init-sources.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class InitSources {
     public static readonly PREFIX_PACKAGES_EXTENSIONS = '@che-';
 
     public static readonly DEFAULT_EXTENSIONS_URI =
-        'https://raw.githubusercontent.com/eclipse-che/che-theia/master/che-theia-init-sources.yml';
+        'https://raw.githubusercontent.com/eclipse-che/che-theia/master-main/che-theia-init-sources.yml';
```
Comment (azatsarynnyy) [inline generator/src/init-sources.ts:35]: This change is to pass the tests. I'll update that to `main` after merging this PR.

**Comment by azatsarynnyy** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by azatsarynnyy** [comment]
[crw-ci-test]

**Comment by azatsarynnyy** [comment]
[crw-ci-test --rebuild]

**Comment by azatsarynnyy** [comment]
[crw-ci-test]

**Comment by azatsarynnyy** [comment]
The HappyPath tests are failed due to the known problem.
See https://www.eclipse.org/lists/che-dev/msg04194.html
I'm going to merge the PR as the proposed changes are related to build, release and docs. But not related to Che-Theia end-user functionality.
We need to merge these changes ASAP to finish switching the default branch from `master` to `main`.

**Review by ericwill — APPROVED** [review]

File: generator/src/init-sources.ts (hunk 1 above)
Comment (azatsarynnyy) [reply generator/src/init-sources.ts:35]: https://github.com/eclipse-che/che-theia/pull/1149

**Pull request merged**